cpp/include/cudf/detail/iterator.cuh

```diff
@@ -313,6 +315,74 @@ struct scalar_pair_accessor : public scalar_value_accessor<Element> {
   }
 };
 
+/**
+ * @brief Utility to discard template type arguments.
+ *
+ * Substitute for std::void_t.
+ *
+ * @tparam T Ignored template parameter
+ */
+template <typename... T>
+using void_t = void;
+
+/**
+ * @brief Compile-time reflection to check if `Element` type has a `rep()` member.
+ */
+template <typename Element, typename = void>
+struct has_rep_member {
+  static constexpr bool value = false;
+};
+
+template <typename Element>
+struct has_rep_member<Element, void_t<decltype(std::declval<Element>().rep())>> {
+  static constexpr bool value = true;
+};
+
+/**
+ * @brief Pair accessor for scalar's representation value and validity.
+ *
+ * @tparam Element The type of element in the scalar.
+ */
+template <typename Element>
+struct scalar_representation_pair_accessor : public scalar_value_accessor<Element> {
+  using super_t    = scalar_value_accessor<Element>;
```

Comment:
`base` instead of `super_t` would be more canonical with C++.

Reply:
Renamed to `base`.